Packages/Status/Sources/Status/Row/StatusRowViewModel.swift

```diff
@@ -37,6 +37,36 @@ public class StatusRowViewModel: ObservableObject {
   @Published var localStatusId: String?
   @Published var localStatus: Status?
 
+  // used by the button to expand a collapsed post
+  @Published var isCollapsed: Bool = true {
+    didSet {
+      recalcCollapse()
+    }
+  }
+  // number of lines to show, nil means show the whole post
+  @Published var lineLimit: Int? = nil
+  // post length determining if the post should be collapsed
+  let collapseThresholdLength : Int = 750
+  // number of text lines to show on a collpased post
+  let collapsedLines: Int = 8
+  var _collapseLongPosts: Bool = false
+  var collapseLongPosts: Bool {
+    get {
+      _collapseLongPosts
+    }
+    set {
+      _collapseLongPosts = newValue
+      recalcCollapse()
+    }
+  }
+  
+  private func recalcCollapse() {
+    let hasContentWarning = !status.spoilerText.asRawText.isEmpty
+    let showCollapseButton = collapseLongPosts && isCollapsed && !hasContentWarning
+      && (status.reblog?.content ?? status.content).asRawText.unicodeScalars.count > collapseThresholdLength
+    lineLimit = showCollapseButton && isCollapsed ? collapsedLines : nil
```

Comment:
This is basically the problem. It's called every time the row appears, triggering an update of a published property as the view appear.

Reply:
is the update triggered even when the published property does not change? Then only setting the property if it really changed should make it better?

Reply:
Yep. You can see the setter in the time profiler. Even if it doens't change anything the body shadow will be recalculated unfortunately.

Reply:
I removed the onAppear an onChange from the view, the user pref is read once in the init() of the model, and the published lineLimit is only set when it changed from the previous value